app/models/claims/claim.rb

```diff
@@ -47,6 +47,7 @@ class Claims::Claim < ApplicationRecord
 
   scope :visible, -> { where(status: VISIBLE_STATUSES) }
   scope :order_created_at_desc, -> { order(created_at: :desc) }
+  scope :updated_over_2_hours_ago, -> { where("updated_at <= ?", 2.hours.ago) }
```

Comment:
Does this really warrant a scope? 🤔 It seems to be very implementation specific.

What happens if you change your definition of a "stale" claim to be > 5 hours ago, rather than 2?

I would suggest either naming the scope something more meaningful and less tightly coupled – for example "stale". Or alternatively just move that query into the service object since that's the only place it's useful.

Reply:
Also there's a slightly more Railsy way to perform this query. Active Record supports [range conditions](https://guides.rubyonrails.org/active_record_querying.html#range-conditions).

Therefore, the query could be written as:

```suggestion
  scope :updated_over_2_hours_ago, -> { where(updated_at: ..2.hours.ago) }
```

which requires no SQL snippets to be written.


The SQL query itself is still the same:

```ruby
Claims::Claim.where(updated_at: ..2.hours.ago).to_sql
```

produces:

```sql
SELECT "claims".* FROM "claims" WHERE "claims"."updated_at" <= '2024-04-04 07:21:53.320273'
```